quic/s2n-quic-core/src/path/mtu.rs

```diff
@@ -273,11 +524,31 @@ impl Controller {
         path_id: path::Id,
         publisher: &mut Pub,
     ) {
-        // MTU probes are only sent in application data space
-        ensure!(packet_number.space().is_application_data());
+        // MTU probes are only sent in the application data space, but since early packet
+        // spaces will use the `InitialMtu` prior to MTU probing being enabled, we need
+        // to check for potentially MTU-related packet loss even when MTU probing is disabled
+        ensure!(self.state.is_disabled() || packet_number.space().is_application_data());
 
         match &self.state {
-            State::Disabled => {}
+            State::Disabled => {
+                if lost_bytes > self.base_plpmtu && self.plpmtu > self.base_plpmtu {
+                    // MTU probing hasn't been enabled yet, but since the initial MTU was configured
+                    // higher than the base PLPMTU and this setting resulted in a lost packet
+                    // we drop back down to the base PLPMTU.
+                    self.plpmtu = self.base_plpmtu;
```

Comment:
Not blocking for this but I wonder if we want to have a threshold on reverting here? Like if 3 packets are lost we then revert. That would avoid artificially clamping the MTU if that first packet was lost but the rest go through.

Reply:
I went with conservatively dropping down to the base just to avoid delaying the completion of the handshake any further. Since the MTU probing will start up again as soon as the handshake completes, that period of time when the MTU is clamped too low didn't seem too bad, though we could revisit this if necessary.

Reply:
yep works for me